pvlib/modelchain.py

```diff
@@ -316,6 +317,10 @@ def __init__(self, system, location,
         self.losses_model = losses_model
         self.orientation_strategy = orientation_strategy
 
+        self.weather = pd.DataFrame()
```

Comment:
Why make this a DataFrame instead of None (like the values below)? I'm not sure if one way is better than the other, but consistency might be more important.